realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt

```diff
@@ -123,9 +124,76 @@ class SessionTests {
         }
     }
 
+    // Check that a Seamless Client Reset is correctly reported.
+    @Test
+    fun errorHandler_discardUnsyncedChangesStrategyReported() = looperThread.runBlocking {
+        val counter = AtomicInteger()
+
+        val incrementAndValidate = {
+            if(2 == counter.incrementAndGet()) {
+                looperThread.testComplete()
+            }
+        }
+
+        val config = SyncConfiguration.Builder(user, "e873fb25-11ef-498f-9782-3c8e1cd2a12c")
+                .assetFile("synced_realm_e873fb25-11ef-498f-9782-3c8e1cd2a12c_no_client_id.realm")
+                .setSyncClientResetStrategy(object: SyncSession.DiscardUnsyncedChangesStrategy{
+                    override fun onBeforeReset(before: Realm, after: Realm) {
```

Comment:
The `onBefore` callback provides the `local` and `remote` Realms before the reset happens, they are named like that by core. I tried to find some better naming, but looks like its more confusing 🙈.

Will revert to the original naming then.

Reply:
But I'm still wondering...why even expose the remote Realm? To me, it just seems confusing because you cannot use it for anything? 

It is nice to pass in the existing Realm, just because it also contains the configuration and thus the path.

Reply:
I guess that allows to report how many changes have been deleted? I agree that it would be more useful to be able to operate with the after Realm, but then wouldn't it be be some sort of a manual reset?

Reply:
There is no easy way to find these objects, but even if you wanted to. Wouldn't it make sense to do it in the `after` callback?

Personally, the API that makes the most sense to me is something like this:

```
override fun onBeforeReset(before: Realm) {

}

override fun onAfterReset(before: Realm, after: Realm) {

}
```

Where the `after` Realm was writeable. That way, you could use the first callback to show something in the UI and the last callback to manually move objects (if you want to) and otherwise just ignore it.

But maybe @ironage had some use cases in mind I don't see for the current API?